pkg/registry/apps/statefulset/strategy_test.go

```diff
@@ -418,21 +440,57 @@ func TestDropStatefulSetDisabledFields(t *testing.T) {
 			oldSS:                 generateStatefulSetWithMinReadySeconds(0),
 			expectedSS:            generateStatefulSetWithMinReadySeconds(10),
 		},
+		{
+			name:                 "MaxUnavailable not enabled, field not used",
+			enableMaxUnavailable: false,
+			ss:                   makeStatefulSetWithMaxUnavailable(nil),
+			oldSS:                nil,
+			expectedSS:           makeStatefulSetWithMaxUnavailable(nil),
+		},
+		{
+			name:                 "MaxUnavailable not enabled, field used in new, not in old",
+			enableMaxUnavailable: false,
+			ss:                   makeStatefulSetWithMaxUnavailable(&maxUnavailable),
+			oldSS:                nil,
+			expectedSS:           makeStatefulSetWithMaxUnavailable(nil),
+		},
+		{
+			name:                 "MaxUnavailable not enabled, field used in old and new",
+			enableMaxUnavailable: false,
+			ss:                   makeStatefulSetWithMaxUnavailable(&maxUnavailable),
+			oldSS:                makeStatefulSetWithMaxUnavailable(&maxUnavailable),
+			expectedSS:           makeStatefulSetWithMaxUnavailable(&maxUnavailable),
```

Comment:
Using the same pointer everywhere makes me nervous (see earlier comments)

Reply:
just to be clear, you mean old and expected, should use different pointers so that we indeed know we are testing correctly , right ?

Reply:
Every instance should use a different pointer, otherwise a bug that changes the value invalidates many test cases.